### PR TITLE
Fix oracle_evaluation for PhaseOracle

### PR DESCRIPTION
**Pull request by a-matsuo**
### Summary
Fix https://github.com/Qiskit/qiskit-terra/issues/6314

### Details and comments
This PR is to revert a bit string when calling `is_good_state` of `PhaseOracle` for the correct `oracle_evaluation` result.

### CODE REVIEW

File: qiskit/algorithms/amplitude_amplifiers/grover.py [hunk 1]
```diff
@@ -249,7 +249,11 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
                 )
 
             all_circuit_results.append(circuit_results)
-            oracle_evaluation = amplification_problem.is_good_state(top_measurement)
+            if isinstance(amplification_problem.oracle, PhaseOracle):
+                oracle_evaluation = amplification_problem.is_good_state(top_measurement[::-1])
```
Comment (Cryoris) [inline qiskit/algorithms/amplitude_amplifiers/grover.py:253]: This seems a bit volatile. If we come up with new oracles we'll potentially have to add more distinctions here. Can't we make `PhaseOracle` use the same order as Grover? Or should we change the order assumed in Grover?

Comment (a-matsuo) [reply qiskit/algorithms/amplitude_amplifiers/grover.py:253]: I agree with that. Actually, I was thinking to change `PhaseOracle` at first. 
What do you think? @boschmitt If you're ok, I'm thinking to change the bit order for evaluation in `PaseOracle`.

Comment (boschmitt) [reply qiskit/algorithms/amplitude_amplifiers/grover.py:253]: Sure you can change `PhaseOracle`. I still think it's not a good long-term solution, but works well as a quick fix

Comment (a-matsuo) [reply qiskit/algorithms/amplitude_amplifiers/grover.py:253]: Thx! I will change the code in `PhaseOracle` for now.

Comment (Cryoris) [reply qiskit/algorithms/amplitude_amplifiers/grover.py:253]: @a-matsuo Did you have time to move the change to `PhaseOracle`? 🙂

Comment (a-matsuo) [reply qiskit/algorithms/amplitude_amplifiers/grover.py:253]: @Cryoris @woodsp-ibm Sorry, I forgot about this PR. I will work on this!

**Comment by a-matsuo** [comment]
I reversed the bit order in `PhaseOracle.evaluate_bitstring()` instead of in `Grover`. Could you guys review the PR please?

**Review by boschmitt — COMMENTED** [review]
LGTM

**Review by boschmitt — DISMISSED** [review]

**Review by Cryoris — APPROVED** [review]

**Review by woodsp-ibm — APPROVED** [review]

**Pull request merged**